src/pages/api/sepana/update.page.ts

```diff
@@ -0,0 +1,230 @@
+        Object.entries(p).reduce(
+          (acc, [k, v]) => ({
+            ...acc,
+            [k]: BigNumber.isBigNumber(v) ? v.toString() : v, // Store BigNumbers as strings
+          }),
+          {} as ProjectJson,
+        ),
+      )
+      .filter(subgraphProject => {
+        const id = subgraphProject.id
+
+        if (!id) return false
+
+        const sepanaProject = sepanaProjects.find(
+          p => subgraphProject.id === p._source.id,
+        )
+
+        if (!sepanaProject) {
+          idsOfNewProjects.add(id)
+        } else if (sepanaProject._source.metadataResolved === false) {
+          missingMetadataCount++
+        }
+
+        // Deep compare subgraph project with sepana project to find which projects have changed or not yet been stored on sepana
+        // Return true to indicate new data should be fetched for this project in sepana db
+        return (
+          !sepanaProject ||
+          (retryIPFS && !sepanaProject._source.metadataResolved) ||
+          projectKeys.some(k => {
+            const oldVal = sepanaProject?._source[k]
+            const newVal = subgraphProject[k as keyof Project]
+
+            if (oldVal !== newVal) {
+              updatedProperties[id] = {
+                key: k,
+                oldVal: oldVal?.toString(),
+                newVal: newVal?.toString(),
+              }
+              return true
+            }
+            return false
+          })
+        )
+      })
+
+    const lastUpdated = await readProvider.getBlockNumber()
+
+    const promises: Promise<{ error?: string; project: SepanaProjectJson }>[] =
+      []
+
+    for (let i = 0; i < changedSubgraphProjects.length; i++) {
+      // Update metadata & `lastUpdated` for each sepana project
+
+      const p = changedSubgraphProjects[i]
+
+      if (i && i % 100 === 0) {
+        // Arbitrary delay to avoid rate limiting
+        await new Promise(r => setTimeout(r, 750))
+      }
+
+      promises.push(
+        infuraApi
+          .get<ProjectMetadataV5>(openIpfsUrl(p.metadataUri), {
+            responseType: 'json',
+            headers: {
+              Accept: 'application/json',
+              'Content-Type': 'application/json',
+            },
+          })
+          .then(({ data: { logoUri, name, description } }) => ({
+            project: {
+              ...p,
+              name,
+              description,
+              logoUri,
+              lastUpdated,
+              metadataResolved: true,
+            } as SepanaProjectJson,
+          }))
+          .catch(error => ({
+            error,
+            project: {
+              ...p,
+              lastUpdated,
+              metadataResolved: false, // If there is an error resolving metadata from IPFS, we'll flag it as `metadataResolved: false`. We'll try getting it again whenever `retryIPFS == true`.
+            } as SepanaProjectJson,
+          })),
+      )
+    }
+
+    const promiseResults = await Promise.all(promises)
```

Comment:
nit - could consider using `allSettled` to get a list of success and fails rather than the catch

Reply:
good suggestion, though i need to actually use the catch to store different data. in the case that resolving metadata fails, we store what we do have along with a `metadataResolved: false` flag